src/test/scala/com/fulcrumgenomics/illumina/SampleSheetTest.scala

```diff
@@ -84,6 +84,33 @@ class SampleSheetTest extends FlatSpec with Matchers with OptionValues {
     an[Exception] should be thrownBy SampleSheet(testDir.resolve("SampleSheetSampleMissingColumns.csv"))
   }
 
+  it should "throw an exception if sample ids are not unique " in {
+    val sample = Sample(sampleOrdinal=0, sampleId="ID", sampleName="NAMEA", libraryId="LIBRARY")
+    an[IllegalArgumentException] should be thrownBy new SampleSheet(Seq(sample, sample.copy(sampleName="NAMEB")))
+
+    // within a lane
+    val sampleForLane = Sample(sampleOrdinal=0, sampleId="ID", sampleName="NAMEA", libraryId="LIBRARY", lane=Some(1))
+    an[IllegalArgumentException] should be thrownBy new SampleSheet(Seq(sampleForLane, sampleForLane.copy(sampleName="NAMEB")))
+  }
+
+  it should "throw an exception if the combination of sample name and library id are not unique" in {
+    val sample = Sample(sampleOrdinal=0, sampleId="ID1", sampleName="NAME", libraryId="LIBRARY")
+    an[IllegalArgumentException] should be thrownBy new SampleSheet(Seq(sample, sample.copy(sampleId="ID2")))
+
+    // within a lane
+    val sampleForLane = Sample(sampleOrdinal=0, sampleId="ID1", sampleName="NAME", libraryId="LIBRARY", lane=Some(1))
+    an[IllegalArgumentException] should be thrownBy new SampleSheet(Seq(sampleForLane, sampleForLane.copy(sampleId="ID2")))
+  }
+
+  it should "not throw an exception if sample ids are unique and if the combination of sample name and library id are unique" in {
+    val sample = Sample(sampleOrdinal=0, sampleId="IDA", sampleName="NAMEA", libraryId="LIBRARY")
+    new SampleSheet(Seq(sample, sample.copy(sampleId="IDB", sampleName="NAMEB"))).size shouldBe 2
+
+    // within lane
+    val sampleForLane = Sample(sampleOrdinal=0, sampleId="IDA", sampleName="NAMEA", libraryId="LIBRARY", lane=Some(1))
+    new SampleSheet(Seq(sampleForLane, sampleForLane.copy(sampleId="IDB", sampleName="NAMEB"))).size shouldBe 2
+  }
+
```

Comment:
It would be nice to see one test with ~2 samples that have different id/name/library values, but are then specified on multiple lanes (i.e. are replicated on lanes 1 and 2) and that this doesn't throw an exception.